### PR TITLE
Implemented extra SDL functions + build fixes

### PR DESCRIPTION
**Pull request by Keyaku**
Changes:
- Added SDL::GetVersion()
- Added SDL::GetPlatform()
- Added SDL::GetPowerInfo()
- Added SDL::GetCPUSpecs()
- Fixed build on Debian (Linux)
- Made minor Documentation corrections

### CODE REVIEW

File: src/sdl2_extras.c [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	SpnValue mmx = SPN_SDLBOOL(SDL_HasMMX());
+	SpnValue rdtsc = SPN_SDLBOOL(SDL_HasRDTSC());
+	SpnValue sse = SPN_SDLBOOL(SDL_HasSSE());
+	SpnValue sse2 = SPN_SDLBOOL(SDL_HasSSE2());
+	SpnValue sse3 = SPN_SDLBOOL(SDL_HasSSE3());
+	SpnValue sse41 = SPN_SDLBOOL(SDL_HasSSE41());
+	SpnValue sse42 = SPN_SDLBOOL(SDL_HasSSE42());
+
+	// Assembling all this mess
+	spn_hashmap_set_strkey(cpu_features, "Cache (KB)", &cache);
```
Comment (H2CO3) [inline src/sdl2_extras.c]: Do we really need to have the `(KB)` and `(MB)` comments? These are keys, after all – if they were struct fields, would you still call them `cache_KB` and `RAM_MB`? This looks like Hungarian notation.

Also, "kilobyte" is lowercase `k`, as in `kB`.

File: src/sdl2_extras.c [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	switch (state) {
+		case SDL_POWERSTATE_ON_BATTERY:
+			return "not plugged in, running on the battery";
+		case SDL_POWERSTATE_NO_BATTERY:
+			return "plugged in, no battery available";
+		case SDL_POWERSTATE_CHARGING:
+			return "plugged in, charging battery";
+		case SDL_POWERSTATE_CHARGED:
+			return "plugged in, battery charged";
+		case SDL_POWERSTATE_UNKNOWN:
```
Comment (H2CO3) [inline src/sdl2_extras.c]: please insert `// FALLTHRU` comment for explicitness.

Furthermore, switch `case`s should not have an extra level of indentation.

File: src/sdl2_extras.c [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+}
+
+/////////////////////////////////
+////    Platform and CPU    /////
+/////////////////////////////////
+int spnlib_SDL_GetPlatform(SpnValue *ret, int argc, SpnValue *argv, void *ctx)
+{
+	// Simplest function to bind
+	*ret = spn_makestring_nocopy(SDL_GetPlatform());
+
```
Comment (H2CO3) [inline src/sdl2_extras.c]: this poor, lonely empty line would rather not live here.

File: src/sdl2_extras.c (hunk 1 above)
Comment (Keyaku) [inline src/sdl2_extras.c]: You are absolutely right. I was conflicted when I first wrote this, so I'll remove those details (as well as lowercase everything and explicitly mention these kB/MB details in the documents)

File: src/sdl2_extras.c (hunk 2 above)
Comment (Keyaku) [inline src/sdl2_extras.c]: Consider it done

File: src/sdl2_extras.c (hunk 3 above)
Comment (Keyaku) [inline src/sdl2_extras.c]: I don't get this one…

File: src/sdl2_extras.c (hunk 1 above)
Comment (H2CO3) [inline src/sdl2_extras.c]: awesome, thanks!

File: src/sdl2_extras.c (hunk 2 above)
Comment (H2CO3) [inline src/sdl2_extras.c]: thanks!

File: src/sdl2_extras.c (hunk 3 above)
Comment (H2CO3) [inline src/sdl2_extras.c]: there's an unnecessary empty line between setting the Sparkling return value and the return from the C function.

Comment (Keyaku) [inline src/sdl2_extras.c]: Oh. I didn't even notice that. Will fix it

**Pull request merged**